api/hasura/event-triggers/refund-pending-gift.ts

```diff
@@ -0,0 +1,66 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { gql } from '../../../api-lib/Gql';
+import { EventTriggerPayload } from '../../../api-lib/types';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  // no parsing should be needed here since this data comes straight from
+  // the database and zeus keeps this consistent for us
+  const payload: EventTriggerPayload<'pending_token_gifts', 'DELETE'> =
+    req.body;
+  const { sender_id, tokens: tokensToRefund } = payload.event.data.old;
+
+  try {
+    const { users_by_pk } = await gql.q('query')({
+      users_by_pk: [
+        { id: sender_id },
+        {
+          give_token_remaining: true,
+          circle_id: true,
+          name: true,
+        },
+      ],
+    });
+    assert(users_by_pk, `dangling give without user: ${sender_id}`);
```

Comment:
do we have something like sentry or bugsnag set up for this kind of stuff? does it work with the vercel serverless functions? it would be nice.

Reply:
We could/should do something like this: https://docs.sentry.io/platforms/node/

Reply:
We could inject it into the hasura validation middleware so it's easy to deploy haha